ROADMAP.md

```diff
@@ -0,0 +1,92 @@
+* NG - "Next Generation." No version designation.
+
+## NG (Next Generation)
+
+In order for io.js to stay competitive we need to work on the next generation of the platform that can more accurately integrate and reflect the advancements in the language and the ecosystem.
+
+While this constitutes a great leap forward for the platform we will be making this leap without breaking backwards compatibility with the existing ecosystem of modules.
+
+NG will use ES6 modules and will be implementing a new platform and standard library available only to modules using this native new style. Modules written prior to NG using the old CommonJS module syntax will continue to operate against the old API. This is what will allow us to make improvements to the platform without breaking compatibility and still letting future NG based applications benefit from all the modules built today.
+
+# Immediate Priorities
+
+## Debugging and Tracing
+
+Debugging is one of the first things from everyone's mouth, both developer and enterprise, when describing trouble they've had with node.js/io.js.
+
+The goal of io.js' effort is to build a healthy debugging and tracing ecosystem and not to try and build any "silver bullet" features for core (like the domains debacle).
+
+The [Tracing WG](https://github.com/iojs/tracing-wg) is driving this effort:
+
+* AsyncWrap improvements - basically just iterations based on feedback from people using it.
+* async-listener - userland module that will dogfood AsyncWrap as well as provide many often requested debugging features.
+* Tracing
+  * Add tracing support for more platforms (LTTng, etc).
+  * [Unify the Tracing endpoint](https://github.com/iojs/io.js/issues/729).
+  * New Chrome Debugger - Google is working on a version of Chrome's debugger that is without Chrome and can be used with io.js.
+
+## Ecosystem Automation
+
+In order to maintain a good release cadence without harming compatibility we must do a better job of understanding exactly what impact a particular change or release will have on the ecosystem. This requires new automation.
+
+The initial goals for this automation are relatively simple but will create a baseline toolchain we can continue to improve upon.
+
+* Produce a list of modules that no longer build between two release versions.
+* Produce a list of modules that use a particular core API.
+* Produce detailed code coverage data for the tests in core.
+
+## Improve Installation and Upgrades
+
+* Host and maintain registry endpoints (Homebrew, apt, etc).
+* Document installation and upgrade procedures with an emphasis on using nvm or nave for development and our registry endpoints for traditional package managers and production.
+
+## Streams
+
+* Fix all existing compatibility issues.
+* Simplify stream creation to avoid user error.
+* Implement WHATWG Streams interface and identify compatibility issues.
```

Comment:
Well, *someone* should implement a reverse compatible version of WHATWG streams but that shouldn't necessarily be `readable-stream`. I'll modify to the language here.